tests/Resources/Page/OCPDashboard/Pods/Pods.robot

```diff
@@ -132,8 +132,15 @@ Verify Container Image
     Should Be Equal  ${image}  ${expected-image-url}
 
 Search Pod
-    [Documentation]   Return list of pods   pod_start_with = here ypu have to provide starting of pod name
+    [Documentation]   Returns list of pods   pod_start_with = here ypu have to provide starting of pod name
     [Arguments]   ${namespace}  ${pod_start_with}
     ${pod} =  Run  oc get pods -n ${namespace} -o json | jq '.items[] | select(.metadata.name | startswith("${pod_start_with}")) | .metadata.name'
     @{list_pods} =  Split String  ${pod}  \n
     [Return]  ${list_pods}[0]
+
+Kill Pod
+    [Documentation]    Search and kills pod with name including ${pod-name-include} and state = Running
+    [Arguments]    ${namespace}    ${pod-name-includes}
+    ${pod}=    Run    oc get pods -n ${namespace} | grep ${pod-name-includes} | grep Running | awk '{print $1;}'
+    Run    oc exec ${pod} -n ${namespace} -- /bin/kill 1
```

Comment:
This might return multiple pods (e.g. with namespace=redhat-ods-applications and pod-name-includes=jupyter). What happens in that case? Are you killing all pods? only one pod? Which one in that case?

Also, why are you not using `oc delete pod` instead of executing kill PID1 in the container? Isn't that killing only 1 container of the pod? What if there are multiple running containers in the pod?

Reply:
No it no returns multiple pods because we are passing full name of build (eg namespace=redhat-ods-applications and pod-name-includes=s2i-tensorflow-gpu-cuda-11.4.2-notebook-5) and pod name = build name + "-build" so evry time it will return just one name.
For `oc delete pod` I Have to talk with jorge about it.

Reply:
You might be passing a fully qualified name to the keyword, but this is sitting in `Pods.robot` and will surely be used by other tests sooner or later. I think you need to also understand what happens when you are returned multiple pods and how to handle that situation.

Reply:
Ok I got it
So here we can, sort the pods using creation time,fetch latest one and it will delete the latest pod..

Reply:
@rkamble5 is going to add the parameter ${container_name} to this keyword to choose the container to run the command in

We are killing the pod to provoke the build to be in Failed state. I believe using `oc delete pod` the resulting state was different (Cancelled?, I don't remember well)

Reply:
>sort the pods using creation time,fetch latest one and it will delete the latest pod

IMHO that's not a good strategy for a general keyword. If you go ahead with that you should probably rename the keyword to something like `Kill Youngest Pod` and clarify what you are doing in the documentation of the keyword.

>We are killing the pod to provoke the build to be in Failed state. I believe using oc delete pod the resulting state was different (Cancelled?, I don't remember well)

That makes sense then. However then the keyword name should probably not be `Kill Pod` but something like `Kill Container In Pod`, or `Make Pod Go In Failed State`